src/Language/PureScript/CodeGen/JS/Optimizer.hs

```diff
@@ -77,7 +77,8 @@ optimize' js = do
     , inlineOperator (C.prelude, (C.$)) $ \f x -> JSApp f [x]
     , inlineOperator (C.prelude, (C.#)) $ \x f -> JSApp f [x]
     , inlineOperator (C.preludeUnsafe, C.unsafeIndex) $ flip JSIndexer
-    , inlineCommonOperators ]) js
+    , inlineCommonOperators
+    , inlineAppliedArrComposition ]) js
```

Comment:
Finding the best order is tricky in general. I've done it by trial and error in the past, but usually I have issues with things like `runST` and do notation. I don't think there's any issue here though.